### PR TITLE
XRT-2842 feat: updates opc-ua sim scripts and add new one showing how to read/write

### PR DESCRIPTION
**Pull request by JonathanWhiteUK**
Also need merging into 2.2 once approved

### CODE REVIEW

**Comment by JonathanWhiteUK** [comment]
Was tempted to add instructions to set up a OPC-UA browser but we have that in like 3 different places in the examples dir so might be worth just making a folder under server/opc-ua covering it maybe? or could just link to the browser docs

File: Simulators/opc-ua/readWrite.lua [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      variant:setScalar(counter_value * sf_value)
+      out_node:setValue(variant)
+      print("output set to  :",counter_value * sf_value)
+    else
+      print("Scale factor value is nil or counter value is nil.")
+    end
+  else
+    print("Scale factor value variant is nil.")
+  end
+end
```
Comment (dpazj) [inline Simulators/opc-ua/readWrite.lua]: EOF newline

File: Simulators/opc-ua/readWrite.lua [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      local variant = Variant.new(DataType.DOUBLE)
+      variant:setScalar(counter_value * sf_value)
+      out_node:setValue(variant)
+      print("output set to  :",counter_value * sf_value)
+    else
+      print("Scale factor value is nil or counter value is nil.")
+    end
+  else
+    print("Scale factor value variant is nil.")
+  end
```
Comment (dpazj) [inline Simulators/opc-ua/readWrite.lua]: Not to big of a fan of all the (nested) if statements checking for `nil`  as it clutters up the function IMO. Are they all necessary for the scrip to work? If they are necessary group them together e.g
```lua
function Update()
  -- Attempt to get the current value of the counter node
  local counter_variant = Variant.new(DataType.DOUBLE)
  local counter_value_variant = counter_node:getValue()
  local sf_value_variant = sf_node:getValue()
  local counter_value
  if counter_value_variant == nil or sf_value_variant == nil then
    return
  end
 //update nodes

end

```

**Review by dpazj — CHANGES_REQUESTED** [review]

File: Simulators/opc-ua/readWrite.lua (hunk 2 above)
Comment (JonathanWhiteUK) [reply Simulators/opc-ua/readWrite.lua]: They are not required, just added them to ensure the variant is returned successfully

can remove

Comment (JonathanWhiteUK) [reply Simulators/opc-ua/readWrite.lua]: removed nil check

**Review by dpazj — APPROVED** [review]
LGTM

**Review by dpazj — APPROVED** [review]
LGTM

**Pull request merged**